ledger/internal/apptxn_test.go

```diff
@@ -3058,3 +3058,222 @@ check:
 	txns(t, l, eval, &fundA, &callA)
 	endBlock(t, l, eval)
 }
+
+func TestForeignAppAccountsAccessible(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	genBalances, addrs, _ := ledgertesting.NewTestGenesis()
+	testConsensusRange(t, 32, 0, func(t *testing.T, ver int) {
+		dl := NewDoubleLedger(t, genBalances, consensusByNumber[ver])
+		defer dl.Close()
+
+		appA := txntest.Txn{
+			Type:   "appl",
+			Sender: addrs[0],
+			ApprovalProgram: main(`
+int 3
+int 3
+==
+assert
+`),
+		}
+
+		appB := txntest.Txn{
+			Type:   "appl",
+			Sender: addrs[0],
+			ApprovalProgram: main(`
+itxn_begin
+	int pay;                itxn_field TypeEnum
+	int 100;     		    itxn_field Amount
+	txn Applications 1
+	app_params_get AppAddress
+	assert
+	itxn_field Receiver
+itxn_submit
+`),
+		}
+
+		dl.beginBlock()
+		dl.txgroup("", &appA, &appB)
+		vb := dl.endBlock()
```

Comment:
```suggestion
		vb := dl.fullBlock(&appA, &appB)
```
unless they actually need to be in a txgroup, rather than just run sequentially.